### ISSUE TITLE
detections is None, when training by 'oim' backbone.  OIM.py #152

### ISSUE THREAD

**Issue by HenryZhangJianhe** [DeepAlchemist/deep-person-search#11]
OIM.py #152 detections, detector_losses = self.roi_heads(features, proposals, images.image_sizes, targets)

It seems that you forgot to return result when self.training=True
OIM.py #215

`
        result, losses = [], {}        
        if self.training:
            loss_detection, loss_box_reg = fastrcnn_loss(class_logits, box_regression,labels, regression_targets)

            loss_reid, acc = self.reid_loss(embeddings_, pids)

            losses = dict(loss_classifier=loss_detection,
                          loss_box_reg=loss_box_reg,
                          loss_ide=loss_reid)
        else:
            boxes, scores, embeddings, labels = \
                self._postprocess_detections(class_logits, box_regression, embeddings_,
                                             proposals, image_shapes)
            num_images = len(boxes)
            for i in range(num_images):
                result.append(
                    dict(
                        boxes=boxes[i],
                        labels=labels[i],
                        scores=scores[i],
                        feats=embeddings[i],
                    )
                )
        return result, losses
`

**Comment by deepalchemist** [comment]
Thanks for your proposal. I have updated OIM.py.
